iina/LegacyMigration.swift

```diff
@@ -61,10 +61,10 @@ class LegacyMigration {
 
       // Deserialize & convert legacy pref value to modern string format:
       guard let legacyData = Preference.data(for: legacyKey) else { continue }
-      guard let color = NSUnarchiver.unarchiveObject(with: legacyData) as? NSColor,
+      guard let color = try? NSKeyedUnarchiver.unarchivedObject(ofClass: NSColor.self, from: legacyData),
```

Comment:
This will not work. The archive in question is _not_ keyed, so `NSKeyedUnarchiver` will not be happy. The `LegacyMigration` class is about migrating an old archive stored in user defaults to a new format. So there is no choice here, the code must use the old deprecated `NSUnarchiver` to read the old archive during the migration. In better languages we'd be able to add a pragma to suppress the warning with a comment was to the deprecated method is required. But last I heard the proposals to add such a feature to Swift were turned down.

For the other changes I will have to defer to other IINA developers who know more about UI code than I do.